crates/bevy_ecs/src/system/mod.rs

```diff
@@ -810,6 +810,31 @@ mod tests {
         run_system(&mut world, sys);
     }
 
+    #[test]
+    fn data_set_system() {
+        fn sys(mut _query: Query<DataSet<(&mut A, &A)>>) {}
+        let mut world = World::default();
+        run_system(&mut world, sys);
+    }
+
+    #[test]
+    #[should_panic]
+    fn conflicting_query_with_data_set_system() {
```

Comment:
Could you add a should_panic test with the `Query<DataSet>` as the first system parameter too. The conflict checking can be order dependent.